File: python/String/13_Roman_To_Integer.py
class RomanToInteger(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        roman={'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        output= 0
        n = len(s)
        for i in range(n):
            if i<n-1 and roman[s[i]]<roman[s[i+1]]:
                output-=roman[s[i]]
            else:
                output+=roman[s[i]]
        return output

def main():
    s= input("Enter the roman string : ")
    obj = RomanToInteger()
    print("The integer value is : ", obj.romanToInt(s))


if __name__ == "__main__":
    main()

File: java/String/13_Roman_To_Integer.java
import java.util.Scanner;
import java.util.HashMap;
class RomanToInteger {
    public static int romanToInt(String s) {
        HashMap<Character, Integer> roman = new HashMap<>();
        roman.put('I',1);
        roman.put('V',5);
roman.put('X',10);
roman.put('L',50);
roman.put('C',100);
roman.put('D',500);
roman.put('M',1000);
    int n = s.length();
    int output = 0;
    for(int i = 0 ; i < n ; i ++ ){
        if (i < n-1 && roman.get(s.charAt(i))<roman.get(s.charAt(i+1))){
            output-=roman.get(s.charAt(i));
        }
        else{
            output+=roman.get(s.charAt(i));
        }

    }
    return output;
    }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the Roman String : ");
        String s = sc.nextLine();
        System.out.println("The integer value is : "+ romanToInt(s));
    }
}